File: ios/Plugin/TapsellPlusPlugin.m
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(TapsellPlusPlugin, "TapsellPlus",
           CAP_PLUGIN_METHOD(initialize, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestRewardedAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showRewardedAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestNativeAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showNativeAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestNativeVideoAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showNativeVideoAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestStandardAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showStandardAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(hideStandardAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(requestVastAd, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(showGDPRDialog, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(setUserConsent, CAPPluginReturnPromise);
)

File: ios/Plugin/TapsellPlusPlugin.swift
import Foundation
import Capacitor

/**
 * TapsellPlus is currently only available for Android.
 * This is a placeholder implementation for iOS.
 */
@objc(TapsellPlusPlugin)
public class TapsellPlusPlugin: CAPPlugin {
    
    @objc func initialize(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func requestRewardedAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func showRewardedAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func requestNativeAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func showNativeAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func requestNativeVideoAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func showNativeVideoAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func requestStandardAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func showStandardAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func hideStandardAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func requestVastAd(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func showGDPRDialog(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
    
    @objc func setUserConsent(_ call: CAPPluginCall) {
        call.reject("TapsellPlus is not supported on iOS")
    }
}

